README.md

```diff
@@ -14,9 +16,69 @@ git clone --recursive https://github.com/Team-Isaac-Polito/reseq_ros2.git src
 
 This will copy the contents of this repository inside the `ros2_ws/src` folder.
 
+In VSCode (or your editor of choice) open `ros2_ws`
+
 ## Install dependencies with ROSDEP
 
 In the directory `ros2_ws` with the apt cache updated run:
+
 ```bash
 rosdep install -iy --from-path src --rosdistro humble
-```
\ No newline at end of file
+```
+
+## Development dependencies
+
+Install dependencies to manually inspect and test the environment:
+
+```
+sudo apt install -y can-utils ros-humble-rviz2 python3-flake8 python3-flake8-blind-except python3-flake8-builtins python3-flake8-class-newline python3-flake8-comprehensions python3-flake8-deprecated python3-flake8-docstrings python3-flake8-import-order python3-flake8-quotes curl lsb-release gnupg
+```
+
+## VSCode Development configuration
+
+Install the following extensions:
+
+```
+code --install-extension ms-python.flake8
+code --install-extension ms-python.isort
+code --install-extension charliermarsh.ruff
+```
+
+In `ros_ws`, modify the file `.vscode/settings.json` by adding the following configuration:
+
+```json
+    "flake8.args": ["--config", "src/reseq_ros2/test/flake8.cfg"],
+    "[python]": {
+        "editor.defaultFormatter": "charliermarsh.ruff",
+        "editor.formatOnSave": true,
+        "editor.codeActionsOnSave": {
+            "source.organizeImports": "explicit",
+            "python.sortImports": "explicit"
+        }
+    },
+    "ruff.format.args": [
+        "--config=src/ruff.toml",
+        "--line-length=99"
+    ],
+    "ruff.organizeImports": false
+```
+
+The files should format correctly on save
+
+## Testing 
+
+To execute ROS2 tests, run the following command which generates a more detailed output:
+
+```
+colcon test --event-handlers=console_direct+
+```
+
+## Gazebo Simulator
+Due to problems with wsl, the project uses _Gazebo Harmonic_, which can be installed from the official Gazebo repositories.
+
+```bash
+sudo curl https://packages.osrfoundation.org/gazebo.gpg --output /usr/share/keyrings/pkgs-osrf-archive-keyring.gpg
```

Comment:
This still gives the same "unable to find" error. I installed some necessary tools in the doc before installing Gazebo harmonic and it worked. We can add this:
`
sudo apt-get update
sudo apt-get install curl lsb-release gnupg
`

Reply:
Yes, I added them in the development dependencies, but they can be moved here